.circleci/config.yml

```diff
@@ -74,6 +120,50 @@ jobs:
           name: Submit coverage data to codecov.
           command: bash <(curl -s https://codecov.io/bash)
           when: always
+      - run:
+          name: Print instructions on regeneration failure.
+          command: |
+            # Exit early if no diff was found
+            [[ ! -z ${DIFF_FOUND} ]] || exit 0
+            echo "Regenerating generated source probably caused the build to break, please run 'go run ./util/compile_protos' and fix the build."
+          when: on_fail
+
+  push-generated-source:
+    docker:
+      - image: golang:1.12
+    environment:
+      GO111MODULE: "on"
+    working_directory: /go/src/github.com/googleapis/gapic-showcase
+    steps:
+      - attach_workspace:
+          at: /go
+      - run:
+          name: Check if we are using generated source.
+          command: |
+            git status --porcelain
+            git diff --exit-code --quiet || echo 'export DIFF_FOUND="true"' >> $BASH_ENV
```

Comment:
If it's not a lot of trouble, can we make the diff checking more pointed? Just diff the protos, generated pkgs, etc.

I don't want to close a bunch of PRs & branches because we are updating docs or utility files.

Reply:
Done!